pkg/assembler/graphql/model/nodes.go

```diff
@@ -1230,6 +1230,79 @@ type VulnerabilitySpec struct {
 	NoVuln *bool     `json:"noVuln,omitempty"`
 }
 
+// Edge allows filtering path/neighbors output to only contain a subset of all
+// possible GUAC verbs.
+//
+// Each member of the enum matches with one of the verbs. The naming scheme is
+// converting the CamelCase name to CAPITALS_WITH_UNDERSCORES.
+//
+// Note that each GUAC verb is at the same time both an Edge and a Node. We need
+// to return it as a Node to have it show up in the return type of the queries and
+// we need it to be an edge to allow filtering paths to only consider certain
+// predicates.
+type Edge string
+
+const (
```

Comment:
I think we need to be able to define one for "subject" vs "depends_on" for example, so that we can filter on the dependency vs the dependent (when exploring connected component). Wondering if this needs to expand to "HAS_SLSA:SUBJECT" and "HAS_SLSA:BUILDER" for example?

Reply:
i guess - do we want to do it in this PR or want to merge in one then extend it?

Reply:
So, we don't have edge directions in GraphQL, we can expand on the current enum but we'll have combinatorial explosion.

Backedges in GraphQL SDL are a better solution, but require more work to get in.

Reply:
whats the path forward here?

Reply:
https://github.com/guacsec/guac/pull/681/commits/510c4371b898aeaf1b70018f8c0aa55efa1c0aa9 is how it will look if we define these to be in the form of `struct_link`.

In fact, this is only half of the work, only from noun to the verb. But we need the reverse too, from the verb to the noun.

If we agree to move forward with this patch, I'll define the others. If we agree that this leads to combinatorial explosion and makes UX worse, then I can revert said commit.

Reply:
I think for our purposes, we only need the edge name, not so much the direction, so there shouldn't be too much combinatorial explosion? Our edge name definitions implicit declare the filter types.

Reply:
There is no edge names in GraphQL, we can only define them by linking nodes together. I'll continue on 510c437 then to add the other side (from noun to verb), then

Reply:
Reverse defined too